### PR TITLE
[ICV] - flag to prevent IE from booting MX VPU

### PR DESCRIPTION
**Pull request by DariaMityagina**
An additional option whose purpose is to prevent the booting of the MX VPU when we want to generate some blobfile.

### CODE REVIEW

File: src/tests/functional/plugin/myriad/shared_tests_instances/behavior/infer_request/config.cpp [hunk 1]
```diff
@@ -25,6 +25,8 @@ namespace {
 
             {{InferenceEngine::MYRIAD_ENABLE_FORCE_RESET, CONFIG_VALUE(YES)}},
             {{InferenceEngine::MYRIAD_ENABLE_FORCE_RESET, CONFIG_VALUE(NO)}},
+            {{InferenceEngine::MYRIAD_DENABLE_MX_BOOT, CONFIG_VALUE(YES)}},
+            {{InferenceEngine::MYRIAD_DENABLE_MX_BOOT, CONFIG_VALUE(NO)}},
```
Comment (ilya-lavrenov) [inline src/tests/functional/plugin/myriad/shared_tests_instances/behavior/infer_request/config.cpp:29]: disable or denable?

Comment (DariaMityagina) [reply src/tests/functional/plugin/myriad/shared_tests_instances/behavior/infer_request/config.cpp]: It's 'disable' now) Thanks

File: src/inference/include/ie/vpu/myriad_config.hpp [hunk 2]
```diff
@@ -24,6 +24,8 @@ namespace InferenceEngine {
  */
 DECLARE_VPU_CONFIG(MYRIAD_ENABLE_FORCE_RESET);
 
+DECLARE_VPU_CONFIG(MYRIAD_DISABLE_MX_BOOT);
```
Comment (ilya-lavrenov) [inline src/inference/include/ie/vpu/myriad_config.hpp:27]: please, provide doxygen documentation for this option

Comment (DariaMityagina) [reply src/inference/include/ie/vpu/myriad_config.hpp:34]: Done, thank you.

**Comment by DariaMityagina** [comment]
@AntonDudchenko could you please take a look?

**Comment by DariaMityagina** [comment]
@Maxim-Doronin I fixed what you asked.

**Review by AntonDudchenko — APPROVED** [review]

File: src/inference/src/ie_core.cpp [hunk 3]
```diff
@@ -1560,8 +1560,15 @@ CompiledModel Core::compile_model(const std::shared_ptr<const ov::Model>& model,
...
+    std::string name = deviceName;
+    auto new_config = config;
+    auto it = config.find("MYRIAD_DISABLE_MX_BOOT");
+    if (it != config.end() && it->second == "YES") {
+        name = "CPU";
+        new_config.erase("MYRIAD_DISABLE_MX_BOOT");
+    }
     OV_CORE_CALL_STATEMENT({
-        auto exec = _impl->LoadNetwork(toCNN(model), deviceName, config);
+        auto exec = _impl->LoadNetwork(toCNN(model), name, new_config);
```
Comment (Maxim-Doronin) [inline src/inference/src/ie_core.cpp:1571]: But we need to compile network for myriad in order to get a blob

**Review by Maxim-Doronin — CHANGES_REQUESTED** [review]

File: src/inference/src/ie_core.cpp (hunk 3 above)
Comment (DariaMityagina) [reply src/inference/src/ie_core.cpp]: There is one of the ways to solve this. I added a confirmation to compile_tool (in the configure() function).

**Comment by DariaMityagina** [comment]
@beleiuandrei could you take a look, please?

File: tools/compile_tool/main.cpp [hunk 4]
```diff
@@ -202,10 +202,14 @@ static std::map<std::string, std::string> parseConfigFile(char comment = '#') {
...
 static std::map<std::string, std::string> configure() {
     const bool isMYRIAD = FLAGS_d.find("MYRIAD") != std::string::npos;
-
     auto config = parseConfigFile();
 
     if (isMYRIAD) {
+        if (config[InferenceEngine::MYRIAD_DISABLE_MX_BOOT] == "YES") {
+            // the second "YES" is used to make sure we are working with compile_tool
+            // and not affecting other applications.
+            config[InferenceEngine::MYRIAD_DISABLE_MX_BOOT] = "YES,YES";
```
Comment (Maxim-Doronin) [inline tools/compile_tool/main.cpp:211]: I think it is a redundant action. Please, revert logic to keep only 1 "YES".
It is a private plugin option and a user should understand what he or she does.

File: inference-engine/src/vpu/common/include/vpu/private_plugin_config.hpp [hunk 5]
```diff
@@ -128,6 +128,15 @@ DECLARE_VPU_CONFIG(MYRIAD_FORCE_PURE_TENSOR_ITERATOR);
...
 //
 
+/**
+ * @brief Used to prevent IE from booting MX VPU.
+ * This is a plugin scope option and must be used with the plugin's SetConfig method
+ * The only possible values are:
+ *     CONFIG_VALUE(YES)
+ *     CONFIG_VALUE(NO) (default value)
+ */
+DECLARE_VPU_CONFIG(MYRIAD_DISABLE_MX_BOOT);
```
Comment (Maxim-Doronin) [inline inference-engine/src/vpu/common/include/vpu/private_plugin_config.hpp:138]: As for me, it is better to add "MYRIAD_ENABLE_MX_BOOT" and make "Yes" a default option.
This way is kept for many other myriad config options

**Review by Maxim-Doronin — CHANGES_REQUESTED** [review]

File: inference-engine/src/vpu/common/include/vpu/private_plugin_config.hpp (hunk 5 above)
Comment (DariaMityagina) [reply inference-engine/src/vpu/common/include/vpu/private_plugin_config.hpp]: Done, thanks

File: tools/compile_tool/main.cpp (hunk 4 above)
Comment (DariaMityagina) [reply tools/compile_tool/main.cpp]: Okay, reverted

**Comment by DariaMityagina** [comment]
@beleiuandrei could you take a look, please?

**Review by beleiuandrei — APPROVED** [review]

**Comment by DariaMityagina** [comment]
@Maxim-Doronin hi!
All requested changes have been addressed.

**Review by Maxim-Doronin — APPROVED** [review]

**Comment by DariaMityagina** [comment]
@AntonDudchenko could you please merge this PR?

**Pull request merged**